### ISSUE TITLE
Codebases are not loaded on space dashboard screen

### ISSUE THREAD

**Issue by ljelinkova** [openshiftio/openshift.io#4513]
E2E encountered this issue twice in the last 24h. After creating new application, you can see on the dashboard the newly created pipeline, but you do not see the codebase. 

![dashboard](https://user-images.githubusercontent.com/2078072/47842076-ee36ed00-ddbb-11e8-96ba-be1e79d19ead.png)

I can see the following log entry in the browser log

**Comment by joshuawilson** [comment]
That is really odd. I can't reproduce it on prod.

Can you add the log entry you mentioned?

**Comment by ljelinkova** [comment]
Sorry about the log, I wanted to attach it but than I was not sure if that's relevant and than I forgot :-)

http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-1b-released/966/03-01-afterEach.log

There is one SEVERE entry at the top of it but it is probably before the launcher was actually called.

**Comment by joshuawilson** [comment]
This might be it. You can see the error in red about 3/4 down the page. 
https://ci.centos.org/job/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-released/3456/console

**Comment by andrewazores** [comment]
There's a .png linked in those log artifacts showing that the app launcher failed while pushing to the new git repo, and there was then a 404 on retrieving the information about the (presumably associated) new Space. I'm not clear on the logs in the initial report vs the logs Joshua has linked - the name of the e2e space differs from what I see in the logs, so these appear to be different test runs and potentially unrelated failures. However, if there was a git connection issue, that would explain both why the Codebases widget could not retrieve a list of repos to display as well as why the launcher failed to push to the repo in both cases. Could this be related to the GitHub outage?

**Comment by joshuawilson** [comment]
It might be a github outage issue. We should add error handling for github outages.

I've been trying to reproduce this and / or trace it from the logs. I can't find it. @ldimaggi @ljelinkova if you can provide more info or a way to reproduce it, we can try to fix it. Currently the e2e are passing, so I'm going to downgrade this one.

**Comment by ljelinkova** [comment]
This issue occurred again during weekend, but intermittently, I am not able to reproduce it myself. It is interesting that I've seen it only on 1b cluster.

I'll try to modify e2e tests to create screenshot of the launcher to see if there were any errors. It is strange that pipeline is created so I'll try to wait for the result and maybe Jenkins job log. 

I'll update this issue once I have some new information.

**Comment by ljelinkova** [comment]
I was able to reproduce this issue locally. It was not GitHub related, simply some racing issue. When you create a new application and click "Return to dashboard" before all checks pass, you are redirected to the dashboard but the dashboard is never updated with the new created codebase. 

So there are 2 probles:
1.  "Return to dashboard" button should be disabled until all checks finish. 
2. Dashboard should be updated once new codebase is created.  I've noticed that the codebases widged is subscribed to broadcasted events 'codebaseAdded' but it seems that nobody emits such an event.
